internal/tls/tls.go

```diff
@@ -0,0 +1,146 @@
+package tls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"sync"
+
+	"golang.org/x/sys/unix"
+	"k8s.io/utils/inotify"
+
+	"git.sr.ht/~spc/go-log"
+)
+
+const (
+	notifyEvents = unix.IN_MOVED_TO | unix.IN_CLOSE_WRITE
+)
+
+type TLSConfig struct {
```

Comment:
Considering the idea in #59 where I suggested using a general purpose `main.Config`, I think this kind of logic could work very well within that config variable. It could hold a pointer to a `tls.Config`, run the inotify watchers to watch for changes to its file paths, and update either its own config values or the `tls.Config` variable with new data.